examples/indexing_routines.py

```diff
@@ -0,0 +1,335 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+import argparse
+import datetime
+import gc
+import math
+
+from benchmark import run_benchmark
+
+try:
+    import cunumeric as np
+except ImportError:
+    import legate.numpy as np
+
+
+def compute_diagonal(steps, N, timing):
+    A1 = np.ones((N,), dtype=int)
+    print("measuring diagonal")
+    start = datetime.datetime.now()
+    for step in range(steps):
+        A2 = np.diag(A1)
+        A1 = np.diag(A2)
+    stop = datetime.datetime.now()
+    delta = stop - start
+    total = delta.total_seconds() * 1000.0
+    if timing:
+        flops = 0
+        print("Total Flops:      " + str(flops / 1e9) + " GFLOPS/iter")
+        space = (N * N + N) * np.dtype(int).itemsize / 1073741824
+        print("Total Size:       " + str(space) + " GB")
+        print("Elapsed Time for diagonal: " + str(total) + " ms")
+        average = total / steps
+        print("Average            :     " + str(average) + " ms")
+        print(
+            "bandwidth:               "
+            + str(space * 1000.0 / (average))
+            + " GB/s"
+        )
+    return total
+
+
+def compute_choose(steps, N, timing):
+    print("measuring choose")
+    A1 = np.ones((N,), dtype=int)
+    B = np.arange(N, dtype=int)
+    C1 = B % N
+    C2 = C1.reshape(
+        (
+            int(N / 100),
+            int(100),
+        )
+    )
+    start = datetime.datetime.now()
+    for step in range(steps):
+        C1 = np.choose(C1, A1, mode="clip")
+        C2 = C1.reshape(
+            (
+                int(N / 100),
+                int(100),
+            )
+        )
+        np.choose(C2, A1, mode="wrap")
+    stop = datetime.datetime.now()
+    delta = stop - start
+    total = delta.total_seconds() * 1000.0
+    if timing:
+        flops = 0
+        print("Total Flops:      " + str(flops / 1e9) + " GFLOPS/iter")
+        space = 3 * N * np.dtype(int).itemsize / 1073741824
+        print("Total Size:       " + str(space) + " GB")
+        print("Elapsed Time for choose: " + str(total) + " ms")
+        average = total / steps
+        print("Average            :     " + str(average) + " ms")
+        print(
+            "FLOPS/s:                 "
+            + str(flops / (average * 1e6))
+            + " GFLOPS/s"
+        )
+        print(
+            "bandwidth:               "
+            + str(space * 1000 / (average))
+            + " GB/s"
+        )
+    return total
+
+
+def compute_repeat(steps, N, timing):
+    A2 = np.ones(
+        (
+            N,
+            N,
+        ),
+        dtype=int,
+    )
+    print("measuring repeat")
+    start = datetime.datetime.now()
+    for step in range(steps):
+        R = np.repeat(int(1), N)
+        A2[:] = np.repeat(A2, R, axis=1)
```

Comment:
I think I'd rather do `A2  = np.repeat(...)`. We're not saving on space, because the `np.repeat` call has to create an intermediate anyway, we're just inserting an extra copy to fill `A2[:]`, which will conflate the numbers for repeat.